doc/source/tutorials/basictutorial.rst

```diff
@@ -0,0 +1,656 @@
+put ``first gear`` and where to put ``second gear``. Wouldn't it be handy if you made rule that the first word your instructor says is the
+gear you start in and the second word he says is the gear you end in? Well luckily there's a way to apply that rule.
+This is were ``parameters`` come into play, all functions get called using ``()`` after the function name and inside of the brackets
+you put the parameters. ::
+
+  Function ShiftGear {
+    Parameter StartGear.
+    Parameter EndGear.
+
+    Let go of the gas pedal.
+    Press in the clutch pedal.
+    Shift the gear stick from StartGear to EndGear.
+    Let go of the clutch pedal.
+    Press in the gas pedal.
+  }
+
+As you can see we replaced the blank spaces with variables (parameters are also variables).
+So to go from first gear to second gear you'd use:
+``ShiftGear(first, second)``.
+To go from second to third you'd use:
+``ShiftGear(second, third)``.
+To go from third to second you'd use:
+``ShiftGear(third, second)``.
+
+A working example of a function
+_______________________________
+
+Here's an example of a simple function which works in KOS: ::
+
+  Function OneThroughFivePrint {
+    print 1.
+    print 2.
+    print 3.
+    print 4.
+    print 5.
+    }
+
+Functions can have any name but avoid making functions and variables the same name as this will very likely cause problems.
+A function will do anything that's inside of the curly brackets. To use this function type the following: ::
+
+ OneThroughFivePrint().
+
+This will show: ::
+
+  1
+  2
+  3
+  4
+  5
+
+A more complex function
+________________________
+
+Here's an example of a more complex function which has a parameter and will also work in KOS:
+
+Let's say we're in a perfectly circular orbit around kerbin, we can use the following formula:
+``velocity = (2 * pi * radius) / orbital period``
+(https://en.wikipedia.org/wiki/Circular_motion#Formulas)
+
+Ignore how ``ship:orbit:period`` works for now, that will be discussed in the next chapter. ::
+
+  Function VelocityCalculator {
+    Parameter OrbitHeight.
+
+    set KerbinRadius to 600000.
+    set TotalRadius to OrbitHeight + KerbinRadius.
+    set OrbitalPeriod to ship:orbit:period.
+    print (2 * 3.1416 * TotalRadius) / OrbitalPeriod.
+  }
+
+If you're in a 400 km circular orbit and type: ::
+
+  VelocityCalculator(400000).
+
+Will show your orbital velocity.
+
+Now what if you want to use the velocity for other calculations, is that possible? Yes of course that's possible!
+The ``return`` command is very helpful is these situations. The ``return`` function returns a value, piece of text, boolean etc and ends
+the function it is in. ::
+
+  Function VelocityCalculator {
+    Parameter OrbitHeight.
+
+    set KerbinRadius to 600000.
+    set TotalRadius to OrbitHeight + KerbinRadius.
+    set OrbitalPeriod to ship:orbit:period.
+    return (2 * 3.1416 * TotalRadius) / OrbitalPeriod.
+    print "this will be skipped".
```

Comment:
Maybe explain why it will be skipped.

```
    return (2 * 3.1416 * TotalRadius) / OrbitalPeriod.
    // A return statement ends the function, meaning
    // nothing that comes after it in the function will
    // actually happen.  For example, you'll never see
    // the following line print out, because the return
    // statement ended the function early:
    print "You Will Never See This Print.".
```